### PR TITLE
feat: full transaction test implementation 

### PR DESCRIPTION
**Pull request by Rqnsom**
commits:
```
feat: add tnx_params support for node rest API
feat(kmd): support - get_keys, sign_transaction
feat: add serialization for MultiSig
feat: synth node support for sending raw bytes
feat: full transaction test implementation
```

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -204,7 +204,15 @@ _TODO: Investigate more REST API calls and possibly include above._
 
 ### ZG-CONFORMANCE-012
 
-    Reserved
+    One synthetic node sends a transacion to the node.
+    The node then broadcasts that transcation to all other nodes.
```
Comment (lwojtow) [inline SPEC.md:208]: Typo in `transcation`

File: SPEC.md [hunk 2]
```diff
@@ -204,7 +204,15 @@ _TODO: Investigate more REST API calls and possibly include above._
...
+    Another synthetic node receives the broadcasted transacion.
```
Comment (lwojtow) [inline SPEC.md:209]: Typo in `transacion`

File: SPEC.md [hunk 3]
```diff
@@ -225,4 +233,4 @@ _TODO: Investigate more REST API calls and possibly include above._
         -> UniEnsBlockReq
         <- TopicMsgResp
 
-    Results should be introspected manually to check node's health and responsiveness (latency, throughput) when requesting block data. 
\ No newline at end of file
+    Results should be introspected manually to check node's health and responsiveness (latency, throughput) when requesting block data.
```
Comment (lwojtow) [inline SPEC.md:236]: `... the node's health...`

File: src/setup/kmd/mod.rs [hunk 4]
```diff
@@ -184,6 +188,31 @@ impl Kmd {
 
         Err(anyhow!("the kmd instance is not started"))
     }
+
+    /// Get the list of public keys in the wallet.
+    pub async fn get_keys(&mut self, wallet_handle: String) -> anyhow::Result<ListKeysResponse> {
+        if let Some(rest_client) = &self.rest_client {
```
Comment (lwojtow) [inline src/setup/kmd/mod.rs:194]: Why no `match` here?

File: src/setup/kmd/rest_api/client.rs [hunk 5]
```diff
@@ -75,4 +79,50 @@ impl ClientV1 {
                 )
             })
     }
+
+    /// Get the list of public keys in the wallet.
+    pub async fn get_keys(&self, wallet_handle: String) -> anyhow::Result<ListKeysResponse> {
```
Comment (lwojtow) [inline src/setup/kmd/rest_api/client.rs:84]: I'd rename the arg to `wallet_handle_token`. Clearly, this is what it is.

**Review by lwojtow — APPROVED** [review]
Small typos and nitpicks, other than that OK.

File: SPEC.md (hunk 1 above)
Comment (Rqnsom) [reply SPEC.md:208]: Done

File: SPEC.md (hunk 2 above)
Comment (Rqnsom) [reply SPEC.md]: Done

File: src/setup/kmd/mod.rs (hunk 4 above)
Comment (Rqnsom) [reply src/setup/kmd/mod.rs:197]: I dunno, I find it more readable than this, which is more tightly coupled:
```rust
match self.rest_client {
  Some(rest_client) => rest_client.get_keys(wallet_handle).await,
  _ => Err(anyhow!("the kmd instance is not started")),
}
```

**Review by asmie — APPROVED** [review]

File: src/setup/kmd/rest_api/client.rs (hunk 5 above)
Comment (Rqnsom) [reply src/setup/kmd/rest_api/client.rs]: Done

File: SPEC.md (hunk 3 above)
Comment (Rqnsom) [reply SPEC.md]: Done

**Pull request merged**